src/gui/uithememanager.cpp

```diff
@@ -58,13 +60,35 @@ void UIThemeManager::initInstance()
 UIThemeManager::UIThemeManager()
 {
     const Preferences *const pref = Preferences::instance();
-    if (pref->useCustomUITheme()
-        && !QResource::registerResource(pref->customUIThemePath(), "/uitheme"))
-        LogMsg(tr("Failed to load UI theme from file: \"%1\"").arg(pref->customUIThemePath()), Log::WARNING);
+    bool useCustomUITheme = pref->useCustomUITheme();
+    if (useCustomUITheme
+        && !QResource::registerResource(pref->customUIThemePath(), "/uitheme")) {
+        LogMsg(tr("Failed to load UI theme from file: \"%1\", falling back to system default theme").arg(pref->customUIThemePath()), Log::WARNING);
+
+        useCustomUITheme = false;
+        Preferences::instance()->setUseCustomUITheme(false);
+    }
 
 #if (defined(Q_OS_UNIX) && !defined(Q_OS_MAC))
     m_useSystemTheme = pref->useSystemIconTheme();
+    const char *iconConfigFile = m_useSystemTheme ? "systemiconconfig.json" : "iconconfig.json";
+#else
+    const char *iconConfigFile = "iconconfig.json";
 #endif
+
+    m_iconsDir = useCustomUITheme ? ":uitheme/icons/" : ":icons/";
+
+    QFile iconJsonMap(m_iconsDir + iconConfigFile);
```

Comment:
I'm not sure that's right... Does it make sense to override mapping for "system theme" icons in a custom theme? Shouldn't there just be a three-state option (default-system-custom)? @Chocobo1?

Reply:
>Does it make sense to override mapping for "system theme" icons in a custom theme?

I think not.

>Shouldn't there just be a three-state option (default-system-custom)?

I agree although I haven't checked how the GUI should look like.